openscenario/map_fragment/src/intersection.cpp

```diff
@@ -0,0 +1,277 @@
+  }();
+
+  const auto resolution = [&]() {
+    node.declare_parameter("resolution", default_value::resolution);
+    return node.get_parameter("resolution").as_int();
+  }();
+
+  const auto number_of_lanes_going_left = [&]() {
+    node.declare_parameter("number_of_lanes_going_left", 0);
+    return node.get_parameter("number_of_lanes_going_left").as_int();
+  }();
+
+  const auto number_of_lanes_going_left_and_straight = [&]() {
+    node.declare_parameter("number_of_lanes_going_left_and_straight", 0);
+    return node.get_parameter("number_of_lanes_going_left_and_straight").as_int();
+  }();
+
+  const auto number_of_lanes_going_straight = [&]() {
+    node.declare_parameter("number_of_lanes_going_straight", 0);
+    return node.get_parameter("number_of_lanes_going_straight").as_int();
+  }();
+
+  const auto number_of_lanes_going_in_all_directions = [&]() {
+    node.declare_parameter("number_of_lanes_going_in_all_directions", 0);
+    return node.get_parameter("number_of_lanes_going_in_all_directions").as_int();
+  }();
+
+  const auto number_of_lanes_going_straight_and_right = [&]() {
+    node.declare_parameter("number_of_lanes_going_straight_and_right", 0);
+    return node.get_parameter("number_of_lanes_going_straight_and_right").as_int();
+  }();
+
+  const auto number_of_lanes_going_right = [&]() {
+    node.declare_parameter("number_of_lanes_going_right", 0);
+    return node.get_parameter("number_of_lanes_going_right").as_int();
+  }();
+
+  const auto output_directory = [&]() {
+    node.declare_parameter("output_directory", default_value::directory());
+    return std::filesystem::path(node.get_parameter("output_directory").as_string());
+  }();
+
+  rcpputils::require_true(
+    number_of_lanes_going_left >= 0 && number_of_lanes_going_left_and_straight >= 0 &&
+      number_of_lanes_going_straight >= 0 && number_of_lanes_going_in_all_directions >= 0 &&
+      number_of_lanes_going_straight_and_right >= 0 && number_of_lanes_going_right >= 0,
+    "Parameters specifying number of lanes must not be smaller than zero");
+
+  rcpputils::require_true(
+    number_of_lanes_going_straight == 0 || number_of_lanes_going_in_all_directions == 0,
+    "Parameter number_of_lanes_going_straight cannot be used in conjunction with "
+    "number_of_lanes_going_in_all_directions");
+
+  const auto total_number_of_lanes_per_leg_per_direction =
+    number_of_lanes_going_left + number_of_lanes_going_left_and_straight +
+    number_of_lanes_going_straight + number_of_lanes_going_in_all_directions +
+    number_of_lanes_going_straight_and_right + number_of_lanes_going_right;
+
+  const auto half_leg_width = total_number_of_lanes_per_leg_per_direction * lane_width;
+
+  rcpputils::require_true(
+    turn_radius >= half_leg_width,
+    "Turn radius must not be smaller than half leg width. Actual values:\n"
+    "- Turn radius: " +
+      std::to_string(turn_radius) +
+      "\n"
+      "- Half leg width: " +
+      std::to_string(half_leg_width));
```

Comment:
This is my rule of thumb, but error messages for programs that are not certain to be executed interactively on the terminal should not contain newline characters. For instance, this program is likely to run on Autoware Evaluator in the future, and Autoware Evaluator's error display UI assumes that error messages do not contain line breaks.